acceptance/tests/reports/corrective_change_new_resource.rb

```diff
@@ -0,0 +1,83 @@
+require 'yaml'
+
+test_name "C98092: corrective change new resource" do
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment_with_teardown(master, app_type)
+
+  tmp_file = {}
+  agents.each do |agent|
+    tmp_file[agent.hostname] = agent.tmpfile(tmp_environment)
+  end
+
+  teardown do
+    step 'clean out produced resources' do
+      agents.each do |agent|
+        on(agent, "rm #{tmp_file[agent.hostname]}", :accept_all_exit_codes => true)
```

Comment:
To be safe and to avoid an incomplete rm command, this on(...) statement should be conditional e.g.
_if tmp_file.has_key?(agent.hostname) && tmp_file[agent.hostname] != ''_